src/test/java/seedu/address/model/person/MajorTest.java

```diff
@@ -0,0 +1,77 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class MajorTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Major(null));
+    }
+
+    @Test
+    public void constructor_invalidMajor_throwsIllegalArgumentException() {
+        String invalidMajor = "invalid";
+        assertThrows(IllegalArgumentException.class, () -> new Major(invalidMajor));
+    }
+
+    @Test
+    public void isValidMajor() {
```

Comment:
Based on naming convention for test cases, if you are testing whether single function is able to handle the different conditionals, it should have a test in front of it for future clarity and readibility. 
e.g. (from the website Week 3 topics)
@Test
    public void testStringConversion() {
        assertEquals("4,7", new IntPair(4, 7).toString());
    } 
    
 when the function name is stringConversion(). Likewise, here it should be named testIsValidMajor()